napari/_qt/qt_main_window.py

```diff
@@ -479,6 +480,13 @@ def __init__(self, viewer: 'Viewer', *, show: bool = True):
         viewer.events.theme.connect(self._update_theme)
         viewer.layers.events.connect(self.file_menu.update)
         viewer.events.status.connect(self._status_changed)
+        with contextlib.suppress(IndexError):
+            viewer.cursor.events.position.disconnect(
```

Comment:
Interesting idea!

I guess this means if I do `viewer.cursor.position = ...` in the console/script in non-headless napari, then we won't get the side-effects anymore. It's a bit weird to have this inconsistency, but otherwise I don't mind this.

Reply:
Oh wait, I'm wrong. The connection below will still handle that.

Reply:
see the line below when I again connect the same function, but in throttled version. 

Or what do you mean by side-effect?

Reply:
Though, I think there should be an explanation in the code that tells the reader why we do this reconnection (because we want to limit it in the GUI).